client/src/common/client.ts

```diff
@@ -817,14 +822,14 @@ export abstract class BaseLanguageClient implements FeatureClient<Middleware, La
 		return this._stateChangeEmitter.event;
 	}
 
-	public get outputChannel(): OutputChannel {
+	public get outputChannel(): LogOutputChannel | OutputChannel {
 		if (!this._outputChannel) {
-			this._outputChannel = Window.createOutputChannel(this._clientOptions.outputChannelName ? this._clientOptions.outputChannelName : this._name);
+			this._outputChannel = Window.createOutputChannel(this._clientOptions.outputChannelName ? this._clientOptions.outputChannelName : this._name, { log: true });
```

Comment:
There was some discussion in https://github.com/microsoft/vscode-languageserver-node/issues/1116 that this was not ideal for existing languages that are already implementing their own filtering by severity in another way -- this is creating a new UI that they won't be listening to and might be confusing. I'd suggest that we either:

1. Don't add log: true here, and expect languages that have opted into the new logging support to pass their channel accordingly. (For the C# extension, this is what we're already doing, so this works great for us.)
2. Add the boolean to clientOptions, so languages opt-in that way.

Reply:
I guess this depends on whether a breaking change is wanted here or not. Changing the default here can have the advantage that it encourages extensions to use the new logging utilities that more closely align with VS Code and the LSP Spec. If a breaking change undesired, I think that the first option is reasonable.

Reply:
There was one request to not make the breaking change, and if an extension wants to opt-in it's not very hard to do so. If anything else, it might be worth merging this in _without_ the break and then the break can be discussed separately.

(I'll defer to @dbaeumer if he just wants to take the break or not.)